plugins/inputs/prometheus/prometheus.go

```diff
@@ -244,8 +268,18 @@ func (p *Prometheus) gatherURL(u URLAndAddress, acc telegraf.Accumulator) error
 	return nil
 }
 
+// Start will start the Kubernetes scraping if enabled in the configuration
+func (p *Prometheus) Start(a telegraf.Accumulator) error {
+	if p.KubernetesScraping {
+		return start(p)
+	}
+	return nil
+}
+
+func (p *Prometheus) Stop() {}
```

Comment:
Need to stop the kubernetes pod monitor.  Easiest way to test this is by sending a SIGHUP, the config could be completely different after reload.